### PR TITLE
Unfinished questions should not be in tests.

### PR DESCRIPTION
**Pull request by draganaHrcek**

### CODE REVIEW

File: web/src/test/java/eu/execom/testgenerator/web/QuestionMother.java [hunk 1]
```diff
@@ -14,8 +14,12 @@ private QuestionMother() {
     // not meant to be instantiated
   }
 
-  public static Question freeFormQuestion(Long id, Category category, int points) {
+  public static Question freeFormQuestion(Long id, Category category, int points, boolean isActiv, boolean isValid,
+      boolean isFinished) {
     Question question = new Question();
+    question.setFinished(isFinished);
+    question.setValid(isValid);
+    question.setActive(isActiv);
```
Comment (adamovivan) [inline web/src/test/java/eu/execom/testgenerator/web/QuestionMother.java:22]: Typo

File: core/src/main/java/eu/execom/testgenerator/core/util/ProblemCreatingTestException.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package eu.execom.testgenerator.core.util;
+
+public class ProblemCreatingTestException extends RuntimeException {
+
+  public ProblemCreatingTestException() {
```
Comment (adamovivan) [inline core/src/main/java/eu/execom/testgenerator/core/util/ProblemCreatingTestException.java:5]: Change exception name, for example, NotEnoughQuestionsException

File: web/src/test/java/eu/execom/testgenerator/web/controller/TestControllerTest.java [hunk 3]
```diff
@@ -210,6 +214,20 @@ public void shouldCreateNewTest() throws Exception {
...
+    String token = "token";
+
+    //when
+    when(testService.create(any(), any(), any())).thenThrow(ProblemCreatingTestException.class);
+
+    //then
+    mockMvc.perform(post(URL_PREFIX + "/actions/invite").content(
+        "{\"formulaId\" : 1,\"fullName\" : \"Slavko Stimac\",\"email\" : \"salasumalomritu@gmail.com\"}")
+                                                        .contentType(MediaType.APPLICATION_JSON_UTF8_VALUE))
+           .andExpect(status().isInternalServerError());
```
Comment (adamovivan) [inline web/src/test/java/eu/execom/testgenerator/web/controller/TestControllerTest.java:228]: You should expect a different server response. Please handle your custom exception in the GlobalExceptionHandler class.

File: web/src/test/java/eu/execom/testgenerator/web/integration/TestControllerIntegrationTest.java [hunk 4]
```diff
@@ -74,6 +75,8 @@
   @Autowired
   private CategoryService categoryService;
   @Autowired
+  private QuestionService questionService;
```
Comment (adamovivan) [inline web/src/test/java/eu/execom/testgenerator/web/integration/TestControllerIntegrationTest.java:78]: Do you use questionService?

File: core/src/main/java/eu/execom/testgenerator/core/util/ProblemCreatingTestException.java [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+    super("You don't have enough questions in database to create test.");
```
Comment (stanislavgrujic) [inline core/src/main/java/eu/execom/testgenerator/core/util/ProblemCreatingTestException.java:6]: When you add a message to an exception, you should not address the end user, but rather state the problem which caused the exception - _There are no enough questions in the database to create the test._

**Pull request merged**